addons/l10n_es_edi_verifactu/tests/test_document.py

```diff
@@ -60,6 +60,105 @@ def test_generation_error(self):
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'invoice_date': '2019-01-30',
+            'date': '2019-01-30',
+            'partner_id': self.partner_b.id,
+            'invoice_line_ids': [
+                Command.create({'product_id': self.product_1.id, 'price_unit': 100.0, 'tax_ids': [Command.set(self.tax21_goods.ids)]}),
+                Command.create({'product_id': self.product_1.id, 'price_unit': 200.0, 'tax_ids': [Command.set(self.tax21_services.ids)]}),
+                Command.create({'product_id': self.product_1.id, 'price_unit': 1000.0, 'tax_ids': [Command.set(self.tax10_goods.ids)]}),
+            ],
+        })
+        invoice.action_post()
+
+        self.env['account.move.reversal'].with_company(self.company).create(
+            {
+                'move_ids': [Command.set((invoice.id,))],
+                'date': '2019-02-10',
+                'journal_id': invoice.journal_id.id,
+            }
+        ).reverse_moves(is_modify=True)
+        credit_note = invoice.reversal_move_id
+
+        substitution_move = invoice.l10n_es_edi_verifactu_substitution_move_ids
+        substitution_move.invoice_line_ids[0].price_unit = 50
+        substitution_move.invoice_date = '2019-02-11'
+        substitution_move.action_post()
+        wizard = self.env['account.move.send'].with_context(
+            active_model='account.move',
+            active_ids=substitution_move.ids,
+        ).create({
+            'checkbox_send_mail': False,
+            **({'l10n_es_edi_facturae_checkbox_xml': False} if self.env['ir.module.module']._get('l10n_es_edi_facturae').state == 'installed' else {}),
+        })
+
+        with self._mock_last_document(None), self.assertRaisesRegex(RedirectWarning, r".*There is no Veri\*Factu document for the substituted record\..*"):
+            wizard.action_send_and_print()
+
+        with self._mock_last_document(None):
+            substitution_move._l10n_es_edi_verifactu_create_documents()
+        errors = ["There is no Veri*Factu document for the substituted record.", "There is no Veri*Factu document for the reversal of the substituted record."]
+        expected_record_values = {
+            'l10n_es_edi_verifactu_state': False,
+            'l10n_es_edi_verifactu_warning': self._mock_format_document_generation_errors(errors),
+            'l10n_es_edi_verifactu_warning_level': 'danger',
+        }
+        self.assertRecordValues(substitution_move, [expected_record_values])
+
+        with self._mock_last_document(None):
+            invoice._l10n_es_edi_verifactu_create_documents()
+
+        with self._mock_zeep_registration_operation_certificate_issue(), self.assertRaisesRegex(RedirectWarning, r".*There is no Veri\*Factu document for the reversal of the substituted record\..*"):
+            wizard.action_send_and_print()
+
+        with self._mock_last_document(None):
+            substitution_move._l10n_es_edi_verifactu_create_documents()
```

Comment:
There is no need to mock; the last document should be the one from the invoice as expected / as it should.
```suggestion
        substitution_move._l10n_es_edi_verifactu_create_documents()
```